pkgs/by-name/ms/msedgedriver/package.nix

```
@@ -0,0 +1,55 @@
+{
+  autoPatchelfHook,
+  fetchurl,
+  glib,
+  lib,
+  nspr,
+  nss,
+  stdenv,
+  unzip,
+  xorg,
+}:
+let
+  channel = "stable";
+  version = "129.0.2792.91";
+  hash = "sha256-69+quFNgjioVBtsRSZDPBYXWhtCvX6cZ5TqGhhf5Ht8=";
+in
+stdenv.mkDerivation {
+  pname = "msedgedriver";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://msedgedriver.azureedge.net/${version}/edgedriver_linux64.zip";
+    inherit hash;
+  };
```

`channel` is not used anywhere, can be deleted.
`hash` is only used at one place, it can be moved there.
`version` is used multiple times. For builders like `stdenv.mkDerivation`, the `finalAttrs` pattern is preferred over `let-in`.
```suggestion

stdenv.mkDerivation (finalAttrs: {
  pname = "msedgedriver";
  version = "129.0.2792.91";

  src = fetchurl {
    url = "https://msedgedriver.azureedge.net/${finalAttrs.version}/edgedriver_linux64.zip";
    hash = "sha256-69+quFNgjioVBtsRSZDPBYXWhtCvX6cZ5TqGhhf5Ht8=";
  };
```